### PR TITLE
Allow adding a header by typing /h1, /h2 etc

### PR DESCRIPTION
**Pull request by aristath**
## Description
Currently users can add a header by typing `###` followed by a space.
This PR adds the same behavior for `/h1` to `/h6` followed by enter/return.

## To test:
Typing `/h4` + enter/return should create a new block with `level:4`. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/heading/transforms.js [hunk 1]
```diff
@@ -67,6 +67,16 @@ const transforms = {
 				} );
 			},
 		} ) ),
+		...[ 1, 2, 3, 4, 5, 6 ].map( ( level ) => ( {
+			type: 'enter',
```
Comment (youknowriad) [inline packages/block-library/src/heading/transforms.js:71]: I didn't know about these kinds of transforms. I wonder if there's a way to make them more obvious in the UI but this works for now.

Should we make it case insensitive?

Comment (aristath) [reply packages/block-library/src/heading/transforms.js:71]: > Should we make it case insensitive?

Good point, they should work both for `/h3` and `/H3`. Pushed an additional commit on https://github.com/WordPress/gutenberg/pull/26597/commits/dcf3909f3714b3a233f05cd24696bad5ec61c471 👍

**Comment by aristath** [comment]
The failing test seems to be unrelated to this PR.

**Comment by courane01** [comment]
Related: https://core.trac.wordpress.org/ticket/51187

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Pull request merged**